app/src/main/java/ch/epfl/polybazaar/UI/SalesOverview.java

```diff
@@ -79,15 +83,26 @@ public void onStart() {
         // prepare top menu
         TextView favorites = findViewById(R.id.favoritesOverview);
         favorites.setOnClickListener(v -> {
-            if(checkUserLoggedIn(this)) {
-                displayFavorites(this);
-            };
+            if (checkUserLoggedIn(this)) {
+                AppUser user = getUser();
+                user.getUserData().addOnSuccessListener(authUser -> {
+                    ArrayList<String> favoritesIds = authUser.getFavorites();
+
+                    // the list of favorites of the user is empty
+                    if (favoritesIds == null || favoritesIds.isEmpty()) {
```

Comment:
`favoritesIds` is instantiated with `new Arraylist<>()` so it couldn't be null right?

Reply:
I wrote this part before favoritesIds was instantiated so it made sense back then. However I think this adds more robustness in case we change the way favorites are created within a user someday.